### PR TITLE
Apply rector fixes

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |  9.0.x
| Description?      | Apply rector fixes to reduce code complexity and improve performance
| Type?             | improvement
| Category?         | CO
| BC breaks?        |  no
| Deprecations?     |  no
| How to test?      | Automated tests

Please review it commit by commit. There is no algorythmic changes

### CODE REVIEW

File: src/PrestaShopBundle/Form/Admin/Type/MoneyWithSuffixType.php [hunk 1]
```diff
@@ -71,10 +71,9 @@ public function configureOptions(OptionsResolver $resolver)
      * Can be used with array_walk().
      *
      * @param string $value
-     * @param int $key not used, it's only here to make this method compatible with array_walk
      * @param string $suffix
      */
-    private function applySuffix(&$value, $key, $suffix)
+    private function applySuffix(&$value, $suffix)
```
Comment (kpodemski) [inline src/PrestaShopBundle/Form/Admin/Type/MoneyWithSuffixType.php]: @FabienPapet I wonder if you have seen a comment inside a doc that this param is to work with array_walk?

Comment (FabienPapet) [reply src/PrestaShopBundle/Form/Admin/Type/MoneyWithSuffixType.php]: I Reverted that change

Comment (kpodemski) [reply src/PrestaShopBundle/Form/Admin/Type/MoneyWithSuffixType.php]: thanks, I reviewed the rest and it looks good to me ❤️ very satisfying changes 😎

**Review by kpodemski — APPROVED** [review]

**Review by PululuK — APPROVED** [review]
@FabienPapet très satisfaisant comme PR 👍🏿

**Review by Hlavtox — APPROVED** [review]
You are the boss 🎉

**Review by boherm — APPROVED** [review]

**Comment by boherm** [comment]
Nice! Thanks @FabienPapet ;)

**Comment by paulnoelcholot** [comment]
Hello @FabienPapet,

I tested your PR and it's good for me! :tada: 

Thanks!

Auto test : https://github.com/paulnoelcholot/testing_pr/actions/runs/12357006794

**Review by paulnoelcholot — APPROVED** [review]

**Pull request merged**